src/redux/logging/loggingMiddleware.ts

```diff
@@ -0,0 +1,109 @@
+import { createAsyncThunk } from '@reduxjs/toolkit';
+import { Topic } from 'openspace-api-js';
+
+import { api } from '@/api/api';
+import { LogLevel } from '@/types/enums';
+import { showNotification } from '@/util/logging';
+
+import { onCloseConnection, onOpenConnection } from '../connection/connectionSlice';
+import { AppStartListening } from '../listenerMiddleware';
+
+let topic: Topic | null = null;
+
+/** This `LogLevel` must match OpenSpace/Ghoul LogLevel in
+ * https://github.com/OpenSpace/Ghoul/blob/f02810ad2f77166711f4503060e20745f0d808c6/include/ghoul/logging/loglevel.h#L41
+ */
+enum OpenSpaceLogLevel {
+  AllLogging = 0,
+  Trace = 1,
+  Debug = 2,
+  Info = 3,
+  Warning = 4,
+  Error = 5,
+  Fatal = 6,
+  NoLogging = 7
+}
+
+type LogMessage = {
+  level: OpenSpaceLogLevel;
+  category: string;
+  message: string;
+  timeStamp: string;
+  dateStamp: string;
+};
+
+export const setupSubscription = createAsyncThunk(
+  'logging/setupSubscription',
+  async () => {
+    topic = api.startTopic('errorLog', {
+      event: 'start_subscription',
+      settings: {
+        logLevel: 'Warning'
```

Comment:
Add a comment as to what this means. I.e. that we subscribe to all log messages that are "Warning" or worse. 

And maybe a comment as to why we are not including Info level log messages as of now? These are otherwise the type of messages that are intended for the user, but with the new notification system they would be too intrusive